lib/pubsub_client/null_subscriber.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+module PubsubClient
+  # A null object to act as a subscriber when clients are in dev or test
+  class NullSubscriber
+    def subscribe(*, &block)
+      yield 'message'
```

Comment:
If you change what value is yielded in the production subscriber code, ensure the yielded values here are duck type compatible with that.